### ISSUE TITLE
Streisand-Gateway Certification Request Fail

### ISSUE THREAD

**Issue by WanderingViking** [jlund/streisand#548]
Been attempting to setup Streisand on an AWS instance but when it gets to the TASK [streisand-gateway : Create the certificate request] it fails. Anyone else having this issue or know of a work around. Once it fails no docs are generated. 

OS: Ubuntu 16.04 LTS (Running on VMWare Workstation 12 Pro)
Ansible 2.2.1.0
Python:  2.7.12
Using an AWS IAM with full permissions as the AWS Access Key ID and Secret Access Key

Error as follows:

TASK [streisand-gateway : Create a certificate authority key and certificate] **
changed: [52.52.166.247]

TASK [streisand-gateway : Create the certificate request] **********************
fatal: [52.52.166.247]: FAILED! => {"changed": true, "cmd": ["openssl", "req", "-new", "-nodes", "-days", "1825", "-newkey", "rsa:3072", "-keyout", "/etc/ssl/private/VPN", "Server.key", "-out", "/etc/ssl/certs/VPN", "Server.csr", "-subj", "/C=US/ST=California/L=Malibu/O=Streisand/OU=Streisand Effect Department/CN=52.52.166.247"], "delta": "0:00:00.003849", "end": "2017-03-26 05:48:19.777382", "failed": true, "rc": 1, "start": "2017-03-26 05:48:19.773533", "stderr": "unknown option Server.key\nreq [options] <infile >outfile\nwhere options  are\n -inform arg    input format - DER or PEM\n -outform arg   output format - DER or PEM\n -in arg        input file\n -out arg       output file\n -text          text form of request\n -pubkey        output public key\n -noout         do not output REQ\n -verify        verify signature on REQ\n -modulus       RSA modulus\n -nodes         don't encrypt the output key\n -engine e      use engine e, possibly a hardware device\n -subject       output the request's subject\n -passin        private key password source\n -key file      use the private key contained in file\n -keyform arg   key file format\n -keyout arg    file to send the key to\n -rand file:file:...\n                load the file (or the files in the directory) into\n                the random number generator\n -newkey rsa:bits generate a new RSA key of 'bits' in size\n -newkey dsa:file generate a new DSA key, parameters taken from CA in 'file'\n -newkey ec:file generate a new EC key, parameters taken from CA in 'file'\n -[digest]      Digest to sign with (md5, sha1, md2, mdc2, md4)\n -config file   request template file.\n -subj arg      set or modify request subject\n -multivalue-rdn enable support for multivalued RDNs\n -new           new request.\n -batch         do not ask anything during request generation\n -x509          output a x509 structure instead of a cert. req.\n -days          number of days a certificate generated by -x509 is valid for.\n -set_serial    serial number to use for a certificate generated by -x509.\n -newhdr        output \"NEW\" in the header lines\n -asn1-kludge   Output the 'request' in a format that is wrong but some CA's\n                have been reported as requiring\n -extensions .. specify certificate extension section (override value in config file)\n -reqexts ..    specify request extension section (override value in config file)\n -utf8          input characters are UTF8 (default ASCII)\n -nameopt arg    - various certificate name options\n -reqopt arg    - various request text options", "stdout": "", "stdout_lines": [], "warnings": []}

RUNNING HANDLER [rsyslog : Restart rsyslog] ************************************

RUNNING HANDLER [openconnect : Restart ocserv] *********************************

RUNNING HANDLER [l2tp-ipsec : Restart Libreswan] *******************************

RUNNING HANDLER [l2tp-ipsec : Restart xl2tpd] **********************************

RUNNING HANDLER [dnsmasq : Restart dnsmasq] ************************************

RUNNING HANDLER [openvpn : Restart OpenVPN] ************************************

RUNNING HANDLER [stunnel : Restart stunnel] ************************************

RUNNING HANDLER [shadowsocks : Restart shadowsocks-libev] **********************

RUNNING HANDLER [ssh : Restart SSH] ********************************************

RUNNING HANDLER [tinyproxy : Restart Tinyproxy] ********************************

RUNNING HANDLER [sslh : Restart sslh] ******************************************

RUNNING HANDLER [monit : Restart Monit] ****************************************

RUNNING HANDLER [nginx : Restart Nginx] ****************************************
	to retry, use: --limit @/home/user/streisand/playbooks/amazon.retry

PLAY RECAP *********************************************************************
52.52.166.247              : ok=260  changed=233  unreachable=0    failed=1   
localhost                  : ok=19   changed=8    unreachable=0    failed=0

**Comment by kernelpanek** [comment]
Immediate fix is to name your server without using whitespace (e.g. `VPNServer` or `VPN_Server`)

TL;DR
-------
When the `./streisand` app prompts you with:
```bash
What should the server be named? Press enter for default (streisand).
 [streisand]: 
```
Are you typing `VPN Server`? If so, then I think the issue with your error above is that the `aws_instance_name` variable gets set to a string with spaces in it. When that variable gets to the `TASK [streisand-gateway : Create the certificate request]` it produces the command:
```json
"cmd": ["openssl", "req", "-new", "-nodes", "-days", "1825", "-newkey", "rsa:3072", "-keyout", "/etc/ssl/private/VPN", "Server.key", "-out", "/etc/ssl/certs/VPN", "Server.csr", "-subj", "/C=US/ST=California/L=Malibu/O=Streisand/OU=Streisand Effect Department/CN=52.52.166.247"]
```
The issue is the `-keyout` switch in that command statement is set to `"-keyout", "/etc/ssl/private/VPN"` and the segment after this is `"Server.key"`. The space in your server name is being interpreted as a separate switch to the `openssl` command. This is likely going to cause you other grief because the `streisand_server_name` variable (what your server name input gets set to) is used in so many other places in the playbook. 

If I knew what the author's intentions were with the user input, I could whip up a PR to fix this. Some solutions could be to remove spaces from the user's input so `VPN Server` becomes `VPNServer` or to replace spaces with dashes or underscores so `VPN Server` becomes `VPN_Server`. The least desirable option is to escape all of the locations in the code where the `streisand_server_name` variable is used to generate other values, commands, etc.

**Comment by WanderingViking** [comment]
Well, I feel like an idiot now. Thanks that was the problem, Used to writing anything in when setting up AWS instances that forgot about No spaces in programming. Thanks for the help
